include/boost/geometry/algorithms/sym_difference.hpp

```diff
@@ -23,6 +31,193 @@ namespace detail { namespace sym_difference
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<Geometry1>::value, true
+                    >::value,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<GeometryOut>::value
+                    >::value
+            >::apply(geometry2, geometry1, robust_policy, out, strategy);
+
+        return out;
+    }
+};
+
+template <typename GeometryOut, typename Areal1, typename Areal2>
+struct sym_difference_areal_areal
+{
+    template
+    <
+        typename RobustPolicy,
+        typename OutputIterator,
+        typename Strategy
+    >
+    static inline OutputIterator apply(Areal1 const& areal1,
+                                       Areal2 const& areal2,
+                                       RobustPolicy const& robust_policy,
+                                       OutputIterator out,
+                                       Strategy const& strategy)
+    {
+        typedef geometry::model::multi_polygon
+            <
+                GeometryOut
+            > helper_geometry_type;
+
+        helper_geometry_type diff12, diff21;
+
+        std::back_insert_iterator<helper_geometry_type> oit12(diff12);
+        std::back_insert_iterator<helper_geometry_type> oit21(diff21);
+
+        geometry::dispatch::intersection_insert
+            <
+                Areal1,
+                Areal2,
+                GeometryOut,
+                overlay_difference,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<Areal1>::value
+                    >::value,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<Areal2>::value, true
+                    >::value
+            >::apply(areal1, areal2, robust_policy, oit12, strategy);
+
+        geometry::dispatch::intersection_insert
+            <
+                Areal2,
+                Areal1,
+                GeometryOut,
+                overlay_difference,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<Areal2>::value
+                    >::value,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<Areal1>::value, true
+                    >::value,
+                geometry::detail::overlay::do_reverse
+                    <
+                        geometry::point_order<GeometryOut>::value
+                    >::value
+            >::apply(areal2, areal1, robust_policy, oit21, strategy);
+
+        return geometry::dispatch::union_insert
+            <
+                helper_geometry_type,
+                helper_geometry_type,
+                GeometryOut
+            >::apply(diff12, diff21, robust_policy, out, strategy);
+    }
```

Comment:
The section calling `intersection_insert` two times can be found in both `sym_difference_generic` and `sym_difference_areal_areal`. This common code could be placed in one function or `generic` version could be called from `areal_areal` version or the `sym_difference_areal_areal` could be removed and this function calling `intersection_insert` could be called directly inside `dispatch::sym_difference_insert` below.